### DISCUSSION TITLE
Using Edge Side Includes to insert content into <Head> component in new app directory structure.

### DISCUSSION THREAD

**Discussion by anthonymunene** [vercel/next.js#48769] [Help]
### Summary

We have a use case where we need to insert content into the <Head> component. Our architecture follows a micro frontend approach where some of the shared features include serving content/resources like scripts within the <head> element. To achieve this, we utilize Edge Side Includes to fetch the required rendered content/fragments from external sources, thereby minimizing the load on the main application. For instance, the following code snippet illustrates how to bring in content that has been rendered by an application.
<esi:include src="/app/endpoint"></esi:include>

Given the new app directory structure and the subsequent move to composing content within <Head> via the metadata api. Are there plans to support custom elements?

**Comment by mikegeyser** [comment]
Hi folks, can anyone help with this? It's a really big problem for us, and we're heavily invested in NextJS for our microfrontends. We'd like to move one and make use of the new React Server Components. 

Without access to the HTML in the head, to put cross-cutting ESIs in, we're kinda stuck.

Thanks in advance.
